In [119]:
#pip install jupyter

In [120]:
#pip install gmaps

In [121]:
#pip install squarify

In [122]:
#pip install dmba

In [123]:
#pip install scikit-surprise

##### <p> Samuel Wolfe <br> July 23rd, 2023 <br> MSBA 206 <br> DMBA Chapter 8 Homework 8.1 & 8.2 </p>

In [124]:
%matplotlib inline
from pathlib import Path
import pandas as pd
import requests
import io
from sklearn.preprocessing import MinMaxScaler
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split, KFold, cross_val_score
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import accuracy_score
from sklearn.neighbors import NearestNeighbors, KNeighborsClassifier, KNeighborsRegressor
from dmba import regressionSummary, classificationSummary, liftChart, gainsChart
from sklearn.metrics import accuracy_score, confusion_matrix, mean_squared_error
import matplotlib.pylab as plt
import numpy as np

In [125]:
def readFile(url):
    download = requests.get(url).content
    # Reading the downloaded content and turning it into a pandas dataframe
    df = pd.read_csv(io.StringIO(download.decode('utf-8')))
    return df
def statslist(df):
    dfStats = pd.DataFrame({'Mean' : df.mean(numeric_only=True),
            'SD' : df.std(numeric_only=True),
            'Min' : df.min(),
            'Max' : df.max(),
            'Median' : df.median(numeric_only=True),
            })
    return dfStats

# 8.1

#### Loading df
#### Note: With Python 3.11.4 pd.get_dummies() does not default return values as uint8, returns as bool. Have to change post.

In [126]:
dfBank = readFile("https://raw.githubusercontent.com/wolfesamk/MSBA-206/main/dmba/UniversalBank.csv")
dfBank['Education'] = dfBank['Education'].astype('category')
dfBank.columns = dfBank.columns.str.replace(' ', '_')
dfBank = pd.get_dummies(dfBank, prefix_sep='_', drop_first=False)
dfBank['Education_1'] = dfBank['Education_1'].astype('uint8')
dfBank['Education_2'] = dfBank['Education_2'].astype('uint8')
dfBank['Education_3'] = dfBank['Education_3'].astype('uint8')
print(dfBank.shape)
dfBank.head(5)

(5000, 16)


ID  Age  Experience  Income  ZIP_Code  Family  CCAvg  Mortgage  \
0   1   25           1      49     91107       4    1.6         0   
1   2   45          19      34     90089       3    1.5         0   
2   3   39          15      11     94720       1    1.0         0   
3   4   35           9     100     94112       1    2.7         0   
4   5   35           8      45     91330       4    1.0         0   

   Personal_Loan  Securities_Account  CD_Account  Online  CreditCard  \
0              0                   1           0       0           0   
1              0                   1           0       0           0   
2              0                   0           0       0           0   
3              0                   0           0       0           0   
4              0                   0           0       0           1   

   Education_1  Education_2  Education_3  
0            1            0            0  
1            1            0            0  
2            1            0            0  
3            0            1            0  
4            0            1            0

#### Creating training and validation splits.

In [127]:
bank_trainData, bank_validData = train_test_split(dfBank, test_size=0.4, random_state=1)
print(bank_trainData.shape, bank_validData.shape)

(3000, 16) (2000, 16)


## 8.1.a Pivot Table

In [128]:
bank_trainData_8_1_a = bank_trainData[['CreditCard', 'Personal_Loan','Online']]
bank_short_PT_8_1_a = bank_trainData_8_1_a.pivot_table(index=['CreditCard', 'Personal_Loan'],
                    columns=['Online'], aggfunc=len)
bank_short_PT_8_1_a

Online                      0     1
CreditCard Personal_Loan           
0          0              792  1117
           1               73   126
1          0              327   477
           1               39    49

## 8.1.b Probability of 1,1,1

In [129]:
prob111_S = f'P(Loan = 1 | CC = 1 | Online = 1): {(bank_short_PT_8_1_a.iat[3,1]) / (bank_short_PT_8_1_a.iat[2,1] + bank_short_PT_8_1_a.iat[3,1]):.2%}'
prob111_S

'P(Loan = 1 | CC = 1 | Online = 1): 9.32%'

## 8.1.c

In [130]:
bank_trainData_8_1_c_1 = bank_trainData[['Personal_Loan','Online']]
bank_short_PT_8_1_c_1 = bank_trainData_8_1_c_1.pivot_table(index=['Personal_Loan'],
                    columns=['Online'], aggfunc=len, margins=True)
bank_short_PT_8_1_c_1

Online            0     1   All
Personal_Loan                  
0              1119  1594  2713
1               112   175   287
All            1231  1769  3000

In [131]:
bank_trainData_8_1_c_2 = bank_trainData[['CreditCard', 'Personal_Loan']]
bank_short_PT_8_1_c_2 = bank_trainData_8_1_c_2.pivot_table(index=['Personal_Loan'],
                    columns=['CreditCard'], aggfunc=len, margins=True)
bank_short_PT_8_1_c_2

CreditCard        0    1   All
Personal_Loan                 
0              1909  804  2713
1               199   88   287
All            2108  892  3000

#### 8.1.d.i

In [132]:
P_CC1_Loan1_S = f'P(CC = 1 ∣ Loan = 1): {(bank_short_PT_8_1_c_2.iat[1,1]) / (bank_short_PT_8_1_c_2.iat[1,2]):.2%}'
P_CC1_Loan1 = (bank_short_PT_8_1_c_2.iat[1,1]) / (bank_short_PT_8_1_c_2.iat[1,2])
P_CC1_Loan1_S

'P(CC = 1 ∣ Loan = 1): 30.66%'

#### 8.1.d.ii

In [133]:
P_Online1_Loan1_S = f'P(Online = 1 ∣ Loan = 1): {(bank_short_PT_8_1_c_1.iat[1,1]) / (bank_short_PT_8_1_c_1.iat[2,1]):.2%}'
P_Online1_Loan1 = (bank_short_PT_8_1_c_1.iat[1,1]) / (bank_short_PT_8_1_c_1.iat[2,1])
P_Online1_Loan1_S

'P(Online = 1 ∣ Loan = 1): 9.89%'

#### 8.1.d.iii

In [134]:
P_Loan1_LoanAll_S = f'P(Loan = 1) (the proportion of loan acceptors): {(bank_short_PT_8_1_c_2.iat[1,2]) / (bank_short_PT_8_1_c_2.iat[2,2]):.2%}'
P_Loan1_LoanAll = (bank_short_PT_8_1_c_2.iat[1,2]) / (bank_short_PT_8_1_c_2.iat[2,2])
P_Loan1_LoanAll_S

'P(Loan = 1) (the proportion of loan acceptors): 9.57%'

#### 8.1.d.iv

In [135]:
P_CC1_Loan0_S = f'P(CC = 1 ∣ Loan = 0): {(bank_short_PT_8_1_c_2.iat[0,1]) / (bank_short_PT_8_1_c_2.iat[0,2]):.2%}'
P_CC1_Loan0 = (bank_short_PT_8_1_c_2.iat[0,1]) / (bank_short_PT_8_1_c_2.iat[0,2])
P_CC1_Loan0_S

'P(CC = 1 ∣ Loan = 0): 29.64%'

#### 8.1.d.v

In [136]:
P_Online1_Loan0_S = f'P(Online = 1 ∣ Loan = 0): {(bank_short_PT_8_1_c_1.iat[0,1]) / (bank_short_PT_8_1_c_1.iat[0,2]):.2%}'
P_Online1_Loan0 = (bank_short_PT_8_1_c_1.iat[0,1]) / (bank_short_PT_8_1_c_1.iat[0,2])
P_Online1_Loan0_S

'P(Online = 1 ∣ Loan = 0): 58.75%'

#### 8.1.d.vi

In [137]:
P_Loan0_S = f'P(Loan = 0): {(bank_short_PT_8_1_c_2.iat[0,2]) / (bank_short_PT_8_1_c_2.iat[2,2]):.2%}'
P_Loan0 = (bank_short_PT_8_1_c_2.iat[0,2]) / (bank_short_PT_8_1_c_2.iat[2,2])
P_Loan0_S

'P(Loan = 0): 90.43%'

## 8.1.e

In [138]:
P_Bayes_Loan1_CC1_Online1 = f'Naive Bayes P(Loan = 1 ∣ CC = 1, Online = 1): {(P_CC1_Loan1 * P_Online1_Loan1 * P_Loan1_LoanAll)/ ((P_CC1_Loan1 * P_Online1_Loan1 * P_Loan1_LoanAll) + (P_CC1_Loan0 * P_Online1_Loan0 * P_Loan0)):.2%}'
P_Bayes_Loan1_CC1_Online1

'Naive Bayes P(Loan = 1 ∣ CC = 1, Online = 1): 1.81%'

## 8.1.f
#### Looking at the raw values, the value generated from the first pivot table in 8.1.b is higher at 9.32% while in 8.1.e the naive Bayes probability is 1.81%. Knowing that Naive Bayes makes the assumption that all variables are independent (hence naive) I would say the first pivot table solution in 8.1.b is more accurate as it takes into consideration all the values.

## 8.1.e

#### Preparing data

In [139]:
classes = [0,1]
predictors = ['Online','CreditCard']
outcome = ['Personal_Loan']

dfBank = readFile("https://raw.githubusercontent.com/wolfesamk/MSBA-206/main/dmba/UniversalBank.csv")
dfBank['Education'] = dfBank['Education'].astype('category')
dfBank.columns = dfBank.columns.str.replace(' ', '_')
dfBank = pd.get_dummies(dfBank, prefix_sep='_', drop_first=False)
dfBank['Education_1'] = dfBank['Education_1'].astype('uint8')
dfBank['Education_2'] = dfBank['Education_2'].astype('uint8')
dfBank['Education_3'] = dfBank['Education_3'].astype('uint8')

X_dfBank = pd.get_dummies(dfBank, prefix_sep='_', drop_first=False)
Y_dfBank = (dfBank[outcome]).astype(int)

scaler = MinMaxScaler()
scaler.fit(X_dfBank)
X_train, X_valid, y_train, y_valid = train_test_split(X_dfBank, Y_dfBank, test_size=0.40, random_state=1)
X_train = scaler.fit_transform(X_train)
X_valid = scaler.transform(X_valid)

#### Run Naive Bayes

In [140]:
bank_nb = MultinomialNB(alpha=1)
bank_nb.fit(X_train, y_train.Personal_Loan)

# predict probabilities
predProb_train = bank_nb.predict_proba(X_train)
predProb_valid = bank_nb.predict_proba(X_valid)

# predict class membership
y_train_pred = bank_nb.predict(X_train)
y_valid_pred = bank_nb.predict(X_valid)

In [141]:
predProb_valid

array([[0.93745255, 0.06254745],
       [0.98474725, 0.01525275],
       [0.9841384 , 0.0158616 ],
       ...,
       [0.97245357, 0.02754643],
       [0.95499039, 0.04500961],
       [0.94814428, 0.05185572]])

In [142]:
y_valid_pred

array([0, 0, 0, ..., 0, 0, 0])

In [143]:
train_df, valid_df = train_test_split(dfBank, test_size=0.4, random_state=1)

pd.set_option('display.precision', 4)
# probability of flight status
print(train_df['Personal_Loan'].value_counts() / len(train_df))
print()

for predictor in predictors:
    # construct the frequency table
    df = train_df[['Personal_Loan', predictor]]
    freqTable = df.pivot_table(index='Personal_Loan', columns=predictor, aggfunc=len)

    # divide each row by the sum of the row to get conditional probabilities
    propTable = freqTable.apply(lambda x: x / sum(x), axis=1)
    print(propTable)
    print()
pd.reset_option('display.precision')

0    0.9043
1    0.0957
Name: Personal_Loan, dtype: float64

Online              0       1
Personal_Loan                
0              0.4125  0.5875
1              0.3902  0.6098

CreditCard          0       1
Personal_Loan                
0              0.7036  0.2964
1              0.6934  0.3066



In [144]:
# classify a specific loan by searching in the dataset
# for a loan with the same predictor values
df = pd.concat([pd.DataFrame({'actual': y_valid.Personal_Loan, 'predicted': y_valid_pred}),
                pd.DataFrame(predProb_valid, index=y_valid.index)], axis=1)
mask = ((bank_validData.Online == 1) & (bank_validData.CreditCard == 1) &
        (bank_validData.Personal_Loan == 1))
df[mask]

actual  predicted         0         1
348        1          1  0.003955  0.996045
1524       1          1  0.001958  0.998042
1273       1          1  0.001525  0.998475
1038       1          1  0.001732  0.998268
4883       1          1  0.001926  0.998074
1050       1          1  0.001301  0.998699
2809       1          1  0.001673  0.998327
891        1          1  0.003690  0.996310
566        1          1  0.001691  0.998309
2707       1          1  0.001898  0.998102
1499       1          1  0.001759  0.998241
2046       1          1  0.001196  0.998804
772        1          1  0.001546  0.998454
4927       1          1  0.002226  0.997774
4175       1          1  0.005800  0.994200
4585       1          1  0.001096  0.998904
4823       1          1  0.002249  0.997751
4147       1          1  0.001889  0.998111
438        1          1  0.001741  0.998259
3327       1          1  0.001872  0.998128
4008       1          1  0.001654  0.998346
1105       1          1  0.001154  0.998846
2443       1          1  0.001216  0.998784
1237       1          1  0.001056  0.998944
3006       1          1  0.001856  0.998144
3373       1          1  0.000799  0.999201
1672       1          1  0.006647  0.993353
75         1          1  0.001237  0.998763
47         1          1  0.001381  0.998619
1882       1          1  0.007035  0.992965
4980       1          1  0.005157  0.994843
2996       1          1  0.006206  0.993794
4705       1          1  0.010684  0.989316

## 8.1.G I believe, if my results are correct, the resulting probabilities show that the Naive Bayes probability generated in 8.1.e is roughly the same as the probability show above. I think this makes sense because we generated these results using a Naive Bayes method, so the probabilities should match.

# 8.2

#### Importing the data and checking the shape of the df while also checking the values of MAX_SEV_IR.

In [145]:
dfAA = readFile("https://raw.githubusercontent.com/wolfesamk/MSBA-206/main/dmba/accidentsFull.csv")
print(dfAA.shape)
dfAA.MAX_SEV_IR.value_counts()

(42183, 24)


1    20996
0    20721
2      466
Name: MAX_SEV_IR, dtype: int64

#### The INJURY Column is created below.

In [146]:
dfAA['INJURY'] = np.where((dfAA.MAX_SEV_IR == 1) | (dfAA.MAX_SEV_IR == 2),1,0)
print(dfAA.shape)
dfAA.INJURY.value_counts()

(42183, 25)


1    21462
0    20721
Name: INJURY, dtype: int64

## 8.2.a

In [147]:
pred8_2_a = f'P(Injury):{(dfAA.INJURY.value_counts()[1])/ (len(dfAA)):.2%}'
pred8_2_a


'P(Injury):50.88%'

#### Given the probability found above, 50.88%. The prediction should technically be if accident is reported it will have an injury.

## 8.2.b

#### Creating isolated dataframe for easier use.

In [148]:
dfAA_b = dfAA.head(12)[['INJURY','WEATHER_R','TRAF_CON_R']]
response = ['INJURY']
predictors = ['WEATHER_R','TRAF_CON_R']

## 8.2.b.i

In [149]:
dfAA_b_PT_i = dfAA_b.pivot_table(index=predictors,
                    columns=response, aggfunc=len, fill_value=0, margins=True)
dfAA_b_PT_i

INJURY                0  1  All
WEATHER_R TRAF_CON_R           
1         0           1  2    3
          1           1  0    1
          2           1  0    1
2         0           5  1    6
          1           1  0    1
All                   9  3   12

In [150]:
prob = []

## 8.2.b.ii

In [151]:
P_I1_W1_T0_S = f'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R =0): {(((dfAA_b_PT_i.iat[0,1]) / (dfAA_b_PT_i.iat[0,2]))*((dfAA_b_PT_i.iat[0,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[0,2]) / (dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W1_T0 = (((dfAA_b_PT_i.iat[0,1]) / (dfAA_b_PT_i.iat[0,2]))*((dfAA_b_PT_i.iat[0,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[0,2]) / (dfAA_b_PT_i.iat[5,2]))
prob.append(P_I1_W1_T0)
P_I1_W1_T0_S

'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R =0): 66.67%'

In [152]:
P_I1_W1_T1_S = f'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 1): {(((dfAA_b_PT_i.iat[1,1]) / (dfAA_b_PT_i.iat[1,2]))*((dfAA_b_PT_i.iat[1,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[1,2]) / (dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W1_T1 = (((dfAA_b_PT_i.iat[1,1]) / (dfAA_b_PT_i.iat[1,2]))*((dfAA_b_PT_i.iat[1,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[1,2]) / (dfAA_b_PT_i.iat[5,2]))
prob.append(P_I1_W1_T1)
P_I1_W1_T1_S

'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 1): 0.00%'

In [153]:
P_I1_W1_T2_S = f'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 2): {(((dfAA_b_PT_i.iat[2,1]) / (dfAA_b_PT_i.iat[2,2]))*((dfAA_b_PT_i.iat[2,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[2,2]) / (dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W1_T2 = (((dfAA_b_PT_i.iat[2,1]) / (dfAA_b_PT_i.iat[2,2]))*((dfAA_b_PT_i.iat[2,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[2,2]) / (dfAA_b_PT_i.iat[5,2]))
prob.append(P_I1_W1_T2)
P_I1_W1_T2_S

'Exact Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 2): 0.00%'

In [154]:
P_I1_W2_T0_S = f'Exact Bayes P(Injury=yes|WEATHER_R = 2 & TRAF_CON_R = 0): {(((dfAA_b_PT_i.iat[3,1]) / (dfAA_b_PT_i.iat[3,2]))*((dfAA_b_PT_i.iat[3,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[3,2]) / (dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W2_T0 = (((dfAA_b_PT_i.iat[3,1]) / (dfAA_b_PT_i.iat[3,2]))*((dfAA_b_PT_i.iat[3,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[3,2]) / (dfAA_b_PT_i.iat[5,2]))
prob.append(P_I1_W2_T0)
P_I1_W2_T0_S

'Exact Bayes P(Injury=yes|WEATHER_R = 2 & TRAF_CON_R = 0): 16.67%'

In [155]:
P_I1_W2_T0_S = f'Exact Bayes P(Injury=yes|WEATHER_R = 2 & TRAF_CON_R = 1): {(((dfAA_b_PT_i.iat[4,1]) / (dfAA_b_PT_i.iat[4,2]))*((dfAA_b_PT_i.iat[4,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[4,2]) / (dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W2_T0 = (((dfAA_b_PT_i.iat[4,1]) / (dfAA_b_PT_i.iat[4,2]))*((dfAA_b_PT_i.iat[4,2]) / (dfAA_b_PT_i.iat[5,2])))/((dfAA_b_PT_i.iat[4,2]) / (dfAA_b_PT_i.iat[5,2]))
prob.append(P_I1_W2_T0)
P_I1_W2_T0_S

'Exact Bayes P(Injury=yes|WEATHER_R = 2 & TRAF_CON_R = 1): 0.00%'

#### Exact Bayes P(Injury=yes|WEATHER_R = 2 & TRAF_CON_R = 2): This is a 0/0 situation and does not become valid.

## 8.2.b.iii

In [156]:
conditions = [((dfAA_b.WEATHER_R == 1) & (dfAA_b.TRAF_CON_R == 0)),
    ((dfAA_b.WEATHER_R == 1) & (dfAA_b.TRAF_CON_R == 1)),
    ((dfAA_b.WEATHER_R == 1) & (dfAA_b.TRAF_CON_R == 2)),
    ((dfAA_b.WEATHER_R == 2) & (dfAA_b.TRAF_CON_R == 0)),
    ((dfAA_b.WEATHER_R == 2) & (dfAA_b.TRAF_CON_R == 1))]

In [173]:
dfAA_b['PROB_INJURY'] = np.select(conditions, prob)
dfAA_b['INJURY_PRED'] = np.where(dfAA_b.PROB_INJURY > 0.5,1,0)
dfAA_b

INJURY  WEATHER_R  TRAF_CON_R  PROB_INJURY  INJURY_PRED
0        1          1           0     0.666667            1
1        0          2           0     0.166667            0
2        0          2           1     0.000000            0
3        0          1           1     0.000000            0
4        0          1           0     0.666667            1
5        1          2           0     0.166667            0
6        0          2           0     0.166667            0
7        1          1           0     0.666667            1
8        0          2           0     0.166667            0
9        0          2           0     0.166667            0
10       0          2           0     0.166667            0
11       0          1           2     0.000000            0

## 8.2.b.iv

In [158]:
dfAA_b_PT_i

INJURY                0  1  All
WEATHER_R TRAF_CON_R           
1         0           1  2    3
          1           1  0    1
          2           1  0    1
2         0           5  1    6
          1           1  0    1
All                   9  3   12

In [159]:
P_I1_W1_T1_NB_S = f'Naive Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 1): {((dfAA_b_PT_i.iat[0,1])/(dfAA_b_PT_i.iat[0,2])) * ((dfAA_b_PT_i.iat[1,1])/(dfAA_b_PT_i.iat[5,1])) * ((dfAA_b_PT_i.iat[0,2])/(dfAA_b_PT_i.iat[5,2])):.2%}'
P_I1_W1_T1_NB = ((dfAA_b_PT_i.iat[0,1])/(dfAA_b_PT_i.iat[0,2])) * ((dfAA_b_PT_i.iat[1,1])/(dfAA_b_PT_i.iat[5,1])) * ((dfAA_b_PT_i.iat[0,2])/(dfAA_b_PT_i.iat[5,2]))
P_I1_W1_T1_NB_S

'Naive Bayes P(Injury=yes|WEATHER_R = 1 & TRAF_CON_R = 1): 0.00%'

## 8.2.b.v

In [167]:
dfAA_b = dfAA.head(12)[['INJURY','WEATHER_R','TRAF_CON_R']]
X_AA_v = dfAA_b[predictors]
Y_AA_v = dfAA_b[response]
AA_nb = MultinomialNB(alpha=1)
AA_nb.fit(X_AA_v, Y_AA_v.INJURY)

# predict probabilities
predProb_AA_Bv = AA_nb.predict_proba(X_AA_v)

# predict class membership
y_predAA_Bv = AA_nb.predict(X_AA_v)

In [168]:
predProb_AA_Bv

array([[0.73282443, 0.26717557],
       [0.71491738, 0.28508262],
       [0.78177885, 0.21822115],
       [0.7966805 , 0.2033195 ],
       [0.73282443, 0.26717557],
       [0.71491738, 0.28508262],
       [0.71491738, 0.28508262],
       [0.73282443, 0.26717557],
       [0.71491738, 0.28508262],
       [0.71491738, 0.28508262],
       [0.71491738, 0.28508262],
       [0.84843129, 0.15156871]])

In [169]:
y_predAA_Bv

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])

In [174]:
df = pd.concat([pd.DataFrame({'actual': Y_AA_v.INJURY, 'predicted': y_predAA_Bv}),
                pd.DataFrame(predProb_AA_Bv, index=Y_AA_v.index)], axis=1)
df

actual  predicted         0         1
0        1          0  0.732824  0.267176
1        0          0  0.714917  0.285083
2        0          0  0.781779  0.218221
3        0          0  0.796680  0.203320
4        0          0  0.732824  0.267176
5        1          0  0.714917  0.285083
6        0          0  0.714917  0.285083
7        1          0  0.732824  0.267176
8        0          0  0.714917  0.285083
9        0          0  0.714917  0.285083
10       0          0  0.714917  0.285083
11       0          0  0.848431  0.151569

## 8.2.b.v Comparing this output with the one generated in 8.2.b.iii we see that in the earlier calculation we got 2/3 accuracy while in this model we got 0/3 correct.